source/slang/slang-parameter-binding.cpp

```diff
@@ -1517,11 +1612,89 @@ static RefPtr<TypeLayout> processEntryPointVaryingParameterDecl(
+    //
+    // The result of this step is the type layout to use for
+    // our field/parameter `decl` in this context.
+    //
+    auto typeLayout = processEntryPointVaryingParameter(context, type, state, varLayout);
+
+    // For Khronos targets (OpenGL and Vulkan), we need to process
+    // the `[[vk::location(...)]]` and `[[vk::index(...)]]` attributes,
+    // if present.
+    //
+    // TODO: In principle we should *also* be using the data from
+    // `SV_Target<N>` semantics as an equivalent to `location = <N>`
+    // when targetting Vulkan. Right now we are kind of skating by
+    // on the fact that people almost always declare `SV_Target`s
+    // in numerical order, so that our automatic assignment of
+    // `location`s in declaration order coincidentally matches
+    // the `SV_Target` order.
+    //
+    if( isKhronosTarget(context->getTargetRequest()) )
+    {
+        if( auto locationAttr = decl->FindModifier<GLSLLocationAttribute>() )
+        {
+            int location = locationAttr->value;
+
+            int index = 0;
+            if( auto indexAttr = decl->FindModifier<GLSLIndexAttribute>() )
+            {
+                index = indexAttr->value;
+            }
+
+            // TODO: We should eventually include validation that a non-zero
+            // `vk::index` is only valid for fragment shader color outputs.
+
+            // Once we've extracted the data from the attribute(s), we
+            // need to apply it to the `varLayout` for the parameter/field `decl`.
+            //
+            LayoutResourceKind kinds[] = { LayoutResourceKind::VaryingInput, LayoutResourceKind::VaryingOutput };
+            for( auto kind : kinds )
+            {
+                auto typeResInfo = typeLayout->FindResourceInfo(kind);
+                if(!typeResInfo)
+                    continue;
+
+                auto varResInfo = varLayout->findOrAddResourceInfo(kind);
+                varResInfo->index = location;
+
+                // Note: OpenGL and Vulkan represent dual-source color blending
+                // differently from multiple render targets (MRT) at the source
+                // level.
+                //
+                // When using MRT, GLSL (and thus SPIR-V) looks like this:
+                //
+                //      layout(location = 0) vec4 a;
+                //      layout(location = 1) vec4 b;
+                //
+                // When using dual-source blending the GLSL/SPIR-V looks like:
+                //
+                //      layout(location = 0)            vec4 a;
+                //      layout(location = 0, index = 1) vec4 b;
+                //
+                // Thus for a parameter of kind `VaryingOutput` when targetting
+                // GLSL/SPIR-V, we need a way to encode the value that was pased
+                // for `index` on the secondary color output.
+                //
+                // We are already using the `index` field in the `VarLayout::ResourceInfo`
```

Comment:
I didn't see this aspect of space usage change discussed on the space variable comment - which might be useful for maintainers.